### PR TITLE
tests: new test to check interfaces after reboot the system

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This is created to cover this bug https://bugs.launchpad.net/snapd/+bug/1721518

### CODE REVIEW

**Comment by sergiocazzolato** [comment]
@zyga change applied, thanks for reviewing

**Comment by sergiocazzolato** [comment]
@mvo5 @zyga @stolowski, I see the test is failing and aparently it is because of the bug.
After the reboot I see:
+ snap interfaces
error: no interfaces found

**Comment by zyga** [comment]
Hmm, so indeed the test shows that after reboot we have no interfaces  at all:
```
+ snap interfaces
error: no interfaces found
```

File: tests/regression/lp-1721518/task.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+details: |
+    This test checks if the interfaces are listed after the machine
+    is rebooted. The test is also checking the interfaces after a 
+    snap is installed and removed.
+
+execute: |
+    # install test-snapd-tools and reboot
+    if [ "$SPREAD_REBOOT" = 0 ]; then
+        snap install test-snapd-tools
```
Comment (zyga) [inline tests/regression/lp-1721518/task.yaml]: Can you please move `CONNECTED_PATTERN=":core-support +core:core-support-plug"` above and then add the same test after the snap is installed but before we reboot? `snap interfaces | MATCH "$CONNECTED_PATTERN"`

File: tests/regression/lp-1721518/task.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+summary: Check that interfaces are listed after reboot the machine
+
+details: |
+    This test checks if the interfaces are listed after the machine
+    is rebooted. The test is also checking the interfaces after a 
+    snap is installed and removed.
+
```
Comment (zyga) [inline tests/regression/lp-1721518/task.yaml]: Can you please add:

```
debug: |
    snap list
```

**Comment by sergiocazzolato** [comment]
@zyga, I see this when I debug it, does it help?

linode:ubuntu-16.04-64 .../tests/regression/lp-1721518# sudo test-snapd-tools.echo pwpw
cannot perform operation: mount --rbind /dev /tmp/snap.rootfs_DS6k3C//dev: No such file or directory

linode:ubuntu-16.04-64 .../tests/regression/lp-1721518# ls -al /tmp/
total 48
drwxrwxrwt 11 root root 4096 Nov 30 02:31 .
drwxr-xr-x 25 root root 4096 Nov 30 02:16 ..
drwxrwxrwt  2 root root 4096 Nov 30 02:17 .font-unix
drwxrwxrwt  2 root root 4096 Nov 30 02:17 .ICE-unix
drwx------  2 root root 4096 Nov 30 02:30 snap.rootfs_PA6OXy
drwx------  2 root root 4096 Nov 30 02:31 snap.rootfs_PqQb20
drwx------  3 root root 4096 Nov 30 02:17 systemd-private-de4dce0fc42544b18001d55aaec4948c-colord.service-H5suY6
drwx------  3 root root 4096 Nov 30 02:17 systemd-private-de4dce0fc42544b18001d55aaec4948c-systemd-timesyncd.service-Pi8Kft
drwxrwxrwt  2 root root 4096 Nov 30 02:17 .Test-unix
-rw-------  1 root root 2176 Nov 30 02:17 tmp.bnpHqcN6wt
drwxrwxrwt  2 root root 4096 Nov 30 02:17 .X11-unix
drwxrwxrwt  2 root root 4096 Nov 30 02:17 .XIM-unix

**Pull request merged**